docs/modules/ROOT/pages/gsn.adoc

```diff
@@ -0,0 +1,77 @@
+
+This may sound strange at first, but paying for user onboarding is a very common bussiness practice. Lots of money are spent on advertising, free trials, new user discounts, etc., all with the goal of user acquisition. And compared to those, the cost of a couple Ethereum transactions is actually very small.
+
+Additionally, you can leverage the GSN in scenarios where your users pay you off-chain in advance (e.g. via credit card), with each GSN-call deducting from their balance on your system. The possibilities are endless!
+
+=== Should I trust these relays?
+
+You don't need to! The GSN is set up in such a way where it's in the relays' best interest to serve your requests, and there are measures in place to penalize them if they misbeheave. All of this happens automatically, so you can safely start using their services worry-free.
+
+== One contract to coordinate them all
+
+There are many meta-transaction implementations out there, but the GSN has a unique detail that makes it special. Sitting in the middle of the system, a smart contract is charged with keeping track of relays, handling relayed transactions, charging their recipients, and generally ensuring all parties stay honest. This contract is called RelayHub, and there is a _single_ instance of it in the whole network (you don't need to deploy your own!). Think of it as a piece of public infrastructure, for all Ethereum users to benefit from.
+
+One of RelayHub's tasks is relay-coordination: all relays will advertise their services on this contract, and your users will query it to find the relay that best suits their purposes. This is out of scope for this guide however, and is not something you need to worry about when writing a recipient contract. If you want to learn more about sending transactions via relays, head to our [GSNProvider guide]().
+
+== Receiving a relayed call
+
+The contract you will be inheriting from is GSNRecipient. It includes the IRelayRecipient interface, plus a number of useful utilities that handle the low-level details of the GSN, so you can focus on your dApp.
+
+Unlike regular contract function calls, each relayed call has an additional number of steps it must go through. Most of writing a recipient involves handling these, which are designed to provide enough flexibility to cater to all possible use cases. Basic recipients can however safely ignore most of them, while still being secure and sound.
+
+The OpenZeppelin Contracts provide a number of [tried-and-tested approaches]() for you to use out of the box, but you should still have a basic idea of what's going on under the hood.
+
+==== acceptRelayedCall
+
+First, RelayHub will ask your recipient contract if it wants to receive a relayed call. Recall that you will be charged for incurred gas costs by the relay, so you should only accept calls that you're willing to pay for!
```

Comment:
Why would RelayHub ask my recipient contract? We are assuming here that the user knows the meta-tx flow in which the relayer calls into the hub, which in turn calls into the recipient. We may want to make that flow explicit before going into the details of each method.